src/wallet/gtest/test_rpc_wallet.cpp

```diff
@@ -239,3 +240,163 @@ TEST(WalletRPCTests, RPCZsendmanyTaddrToSapling)
     RegtestDeactivateSapling();
     UnloadGlobalWallet();
 }
+
+TEST(WalletRPCTests, ZIP317Fee)
+{
+    LoadProofParameters();
+    SelectParams(CBaseChainParams::TESTNET);
+
+    LoadGlobalWallet();
+
+    RegtestActivateSapling();
+    {
+        LOCK2(cs_main, pwalletMain->cs_wallet);
+
+        if (!pwalletMain->HaveMnemonicSeed()) {
+            pwalletMain->GenerateNewSeed();
+        }
+
+        UniValue retValue;
+
+        KeyIO keyIO(Params());
+        // add keys manually
+        auto taddr = pwalletMain->GenerateNewKey(true).GetID();
+        auto pa = pwalletMain->GenerateNewLegacySaplingZKey();
+
+        const Consensus::Params& consensusParams = Params().GetConsensus();
+
+        int nextBlockHeight = chainActive.Height() + 1;
+
+        // Add a fake transaction to the wallet
+        CMutableTransaction mtx = CreateNewContextualCMutableTransaction(consensusParams, nextBlockHeight, false);
+        CScript scriptPubKey = CScript() << OP_DUP << OP_HASH160 << ToByteVector(taddr) << OP_EQUALVERIFY << OP_CHECKSIG;
+        size_t utxoCount = 100;
+        for (size_t i = 0; i < utxoCount; i++) {
+            mtx.vout.push_back(CTxOut(5 * COIN, scriptPubKey));
+        }
+        CWalletTx wtx(pwalletMain, mtx);
+        pwalletMain->LoadWalletTx(wtx);
+
+        // Fake-mine the transaction
+        EXPECT_EQ(-1, chainActive.Height());
+        CBlock block;
+        block.vtx.push_back(wtx);
+        block.hashMerkleRoot = BlockMerkleRoot(block);
+        auto blockHash = block.GetHash();
+        CBlockIndex fakeIndex {block};
+        mapBlockIndex.insert(std::make_pair(blockHash, &fakeIndex));
+        chainActive.SetTip(&fakeIndex);
+        EXPECT_TRUE(chainActive.Contains(&fakeIndex));
+        EXPECT_EQ(0, chainActive.Height());
+        wtx.SetMerkleBranch(block);
+        pwalletMain->LoadWalletTx(wtx);
+
+        // Add keys manually
+        std::string taddr_string = keyIO.EncodeDestination(taddr);
+
+        WalletTxBuilder builder(Params(), minRelayTxFee);
+
+        auto selector = CWallet::LegacyTransparentZTXOSelector(
+                true,
+                TransparentCoinbasePolicy::Disallow);
+
+        { // test transparent inputs to NetAmountRecipient
+            auto saplingKey = pwalletMain->GenerateNewLegacySaplingZKey();
+            NetAmountRecipient zaddr2(saplingKey, std::nullopt);
```

Comment:
Why `zaddr2`?
```suggestion
            NetAmountRecipient zaddr(saplingKey, std::nullopt);
```

Reply:
Ah, just a remnant of our copy/paste – `zaddr1` is Sprout and `zaddr2` is Sapling in the other test.